### ISSUE TITLE
Dim terminal text should have half the minimum contrast ratio

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode#190371]
See https://github.com/microsoft/vscode/issues/178162 for an example of why this is an issue

Repro:

- Set a high minimum contrast ratio in settings
- Run `echo -e '\x1b[31mNormal \x1b[31;2mDim\x1b[0m'`, 🐛 provided the minimum contrast ratio is not too high, the colors should differ

Actual:

![Image](https://github.com/microsoft/vscode/assets/2193314/11aa5ec6-da63-4eba-8d6b-b125d0d52e69)

Expected:

![Image](https://github.com/microsoft/vscode/assets/2193314/3ca6b31d-c0be-4f05-b7e6-353d50b326ac)

**Comment by Tyriar** [comment]
Upstream: https://github.com/xtermjs/xterm.js/issues/4262
